packages/cli/src/commands/setup/vite/vite.js

```diff
@@ -0,0 +1,154 @@
+import fs from 'fs'
+import path from 'path'
+
+import toml from '@iarna/toml'
+import chalk from 'chalk'
+import { Listr } from 'listr2'
+
+import { addWebPackages } from '@redwoodjs/cli-helpers'
+import { getConfigPath } from '@redwoodjs/internal/dist/paths'
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths, transformTSToJS, writeFile } from '../../../lib'
+import c from '../../../lib/colors'
+import { isTypeScriptProject } from '../../../lib/project'
+
+export const command = 'vite'
+export const description =
+  '[Experimental] Configure the web side to use Vite, instead of Webpack'
+export const builder = (yargs) => {
+  yargs.option('force', {
+    alias: 'f',
+    default: false,
+    description: 'Overwrite existing configuration',
+    type: 'boolean',
+  })
+  yargs.option('verbose', {
+    alias: 'v',
+    default: false,
+    description: 'Print more logs',
+    type: 'boolean',
+  })
+  yargs.option('add-package', {
+    default: true,
+    description:
+      'Allows you to skip adding the @redwoodjs/vite package. Useful for testing',
+    type: 'boolean',
+  })
+}
+
+export const handler = async ({ force, verbose, addPackage }) => {
```

Comment:
The handler and it's dependencies need to be in its own file, `viteHandler.js` (see `builderHandler.js`, `devHandler.js`, etc) and asynchronously imported here; else the upfront require cost is going to slow down other commands

Reply:
We haven't done this with many setup commands before, but I think we should continue the trend

Reply:
Got a PR here to get the other commands on the same page https://github.com/redwoodjs/redwood/pull/7156

Reply:
Yeah happy to change this, no worries! Feel free to push if you like